core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/PoolableElementView.java

```diff
@@ -534,4 +534,172 @@ public void setSample(Sample sample) {
+    result = prime * result + ((dilutionConcentrationUnits == null) ? 0 : dilutionConcentrationUnits.hashCode());
+    result = prime * result + (int) (dilutionId ^ (dilutionId >>> 32));
+    result = prime * result + ((dilutionName == null) ? 0 : dilutionName.hashCode());
+    result = prime * result + ((dilutionNgUsed == null) ? 0 : dilutionNgUsed.hashCode());
+    result = prime * result + ((dilutionVolume == null) ? 0 : dilutionVolume.hashCode());
+    result = prime * result + ((dilutionVolumeUsed == null) ? 0 : dilutionVolumeUsed.hashCode());
+    result = prime * result + ((indices == null) ? 0 : indices.hashCode());
+    result = prime * result + ((lastModified == null) ? 0 : lastModified.hashCode());
+    result = prime * result + ((lastModifierName == null) ? 0 : lastModifierName.hashCode());
+    result = prime * result + ((libraryAlias == null) ? 0 : libraryAlias.hashCode());
+    result = prime * result + ((libraryBarcode == null) ? 0 : libraryBarcode.hashCode());
+    result = prime * result + ((libraryDescription == null) ? 0 : libraryDescription.hashCode());
+    result = prime * result + ((libraryDnaSize == null) ? 0 : libraryDnaSize.hashCode());
+    result = prime * result + ((libraryId == null) ? 0 : libraryId.hashCode());
+    result = prime * result + ((libraryName == null) ? 0 : libraryName.hashCode());
+    result = prime * result + (libraryPaired ? 1231 : 1237);
+    result = prime * result + ((librarySelectionType == null) ? 0 : librarySelectionType.hashCode());
+    result = prime * result + ((libraryStrategyType == null) ? 0 : libraryStrategyType.hashCode());
+    result = prime * result + (lowQualityLibrary ? 1231 : 1237);
+    result = prime * result + ((platformType == null) ? 0 : platformType.hashCode());
+    result = prime * result + ((preMigrationId == null) ? 0 : preMigrationId.hashCode());
+    result = prime * result + ((projectAlias == null) ? 0 : projectAlias.hashCode());
+    result = prime * result + ((projectId == null) ? 0 : projectId.hashCode());
+    result = prime * result + ((projectName == null) ? 0 : projectName.hashCode());
+    result = prime * result + ((projectShortName == null) ? 0 : projectShortName.hashCode());
+    result = prime * result + ((sampleAccession == null) ? 0 : sampleAccession.hashCode());
+    result = prime * result + ((sampleAlias == null) ? 0 : sampleAlias.hashCode());
+    result = prime * result + ((sampleDescription == null) ? 0 : sampleDescription.hashCode());
+    result = prime * result + ((sampleId == null) ? 0 : sampleId.hashCode());
+    result = prime * result + ((sampleName == null) ? 0 : sampleName.hashCode());
+    result = prime * result + ((sampleType == null) ? 0 : sampleType.hashCode());
+    result = prime * result + ((targetedSequencingId == null) ? 0 : targetedSequencingId.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 151 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)